ConditionalCompilation.md

```diff
@@ -0,0 +1,15 @@
+# Conditional Compilation
+
+(TBD)
+
+This section will discuss extensions to WGSL
+to support conditional compilation, where the conditions
+are set by the host code runtime or build time.
+
+* `#if #else #endif` (or perhaps `#ifdef`) in the style of a C preprocessor 
```

Comment:
I'd really prefer the rust style as it's so much more structured. It would make optimizations a lot easier. For example could parse/lower certain functions ahead of time and easily remove them from the IR vs `#if #else` which operates on a source level.

Reply:
yep, adding to discussion session. (also I agree that the c processor style is clunkier and we should consider rust style alternative)